src/Microsoft.Health.Fhir.Liquid.Converter/Filters/CollectionFilters.cs

```diff
@@ -45,10 +45,14 @@ public static string BatchRender(Context context, List<object> collection, strin
             }
 
             var sb = new StringBuilder();
-            collection?.ForEach(entry =>
+            context.Stack(() =>
```

Comment:
why do we need stack the context here?

Reply:
We need to push context and pop after render sub template.